### ISSUE TITLE
Since 1.5.6 the last column doesn't expand with the window width anymore

### ISSUE THREAD

**Issue by MartinX3** [moneymanagerex/moneymanagerex#3814]
MMEX version:
 - [x] 1.5.6
 - [ ] 1.6.x

Operating System:
 - [ ] Windows
 - [ ] Mac OSX
 - [x] Linux 

# Description of the bug

Since 1.5.6 the last column doesn't expand with the window width anymore

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Accountview

Expected result:
All space used

Actual result:
Big, grey area
# Additional information

![grafik](https://user-images.githubusercontent.com/4518092/130666098-6c9fee91-529d-40d1-a0d5-8b2a3d975ebd.png)

PS: If I double-click the column border, it expands as much as the longest text inside this column, which creates a horizontal scrollbar.

**Comment by whalley** [comment]
I've checked (on MacOS) back to 1.3.6 and the behavior for the columns (including the last) is the same. It retains the size you set of it and does not expand to fit available space. Double-clicking a column and expanding to the maximum needed size is the expected behavior.

**Comment by MartinX3** [comment]
Not on 1.5.5 on arch linux (wx 3.0.5).
It must be a wxWidgets difference

![grafik](https://user-images.githubusercontent.com/4518092/130677528-25d5f33c-f162-4aca-8637-0f8758f86785.png)

**Comment by PMaff** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/2698#issuecomment-1083621719

**Comment by MartinX3** [comment]
My problem was not #2698
With wxWidgets 3.0.5 I never had this free space, since the last column was always glued to the right border of the window.
Since 3.1.5 it isn't glued anymore and leaves an empty space.

**Comment by vomikan** [comment]
We can't support wx 3.0.5.
It does not support a lot of features that needed for mmex.

**Comment by MartinX3** [comment]
I understand that
But maybe the behavior of the table element can be restored how it acted in 3.0.x.
It was way more user friendly to enlarge the table by making the window bigger.
Now I need to do 2 steps.
- enlarging the window
- enlarging the column.

**Comment by MartinX3** [comment]
@vomikan will the new wxGTK 3.1.6 fix this broken behavior?

**Comment by MartinX3** [comment]
Weird, it worked for me with mmex v1.5.5 with wxWidgets v3.0.5 on arch linux

**Comment by PMaff** [comment]
It is even more weird. Tried this with 
Version        : 1.5.14-Beta.1
and wxWidget 3.1.5 :

https://user-images.githubusercontent.com/93879362/161768212-8de02bdb-bd1b-4f1f-8e01-211a02cb6097.mp4

So this space behind the last column seems to be only shown in full screen windows mode.

**Comment by MartinX3** [comment]
I don't know how to force a full screen window mode.
I only use the maximize/normal mode icon and drag and drop the window borders.

**Comment by MartinX3** [comment]
Ahhhh
Here the table has the width of the monitor screen where the app window got created the first time.
Switching between maximized and non maximized doesn't change it.

**Comment by PMaff** [comment]
@MartinX3 can you close that ticker?

**Comment by MartinX3** [comment]
Why? The issue is not fixed. If I resize the window I get a large empty space.

**Comment by whalley** [comment]
> The wxWidgets-Changelog lists
> 
> `Generic wxDataViewCtrl now always resizes its last column to fill all the available space, as the GTK+ version always did.`
> 
> But it doesn't work in mmex.

We don't use wxDataViewCtrl.

**Comment by MartinX3** [comment]
> We don't use wxDataViewCtrl.

Oh, that means mmex uses a custom self made table gui element with this bug?

**Comment by vomikan** [comment]
> Oh, that means mmex uses a custom self made table gui element with this bug?

We use wxListCtrl class.

**Comment by vomikan** [comment]
On my Windows system I can hide the Notes column
![image](https://user-images.githubusercontent.com/6836805/164977220-9d943f85-ecd3-462d-949c-b21b4cadb004.png)

then expand it from zero size to any suitable.

**Comment by PMaff** [comment]
> Why? The issue is not fixed. If I resize the window I get a large empty space.

Ok, my misunderstanding.

**Comment by MartinX3** [comment]
I don't know if this happens because of the new wxwidgets version or with mmex 1.5.20 (maybe a previous version as well), but now I've got empty columns I can sort the table with
![image](https://user-images.githubusercontent.com/4518092/188425552-b6dd5542-2dd7-4c79-9b5c-236cd5f3d62d.png)

**Comment by MartinX3** [comment]
@vomikan stale means it will be fixed in 1.6.1?

**Comment by vomikan** [comment]
This issue has not been fixed. But since there are no updates on it, I decided that this is no longer relevant.

I was wrong.

**Comment by MartinX3** [comment]
![image](https://github.com/user-attachments/assets/d5e41c2a-6495-4dd2-a5c4-15426b6244d2)
